src/resumable-subscription.ts

```diff
@@ -1,130 +1,102 @@
-                this.assertState(['OPENING']);
-                this.state = ResumableSubscriptionState.OPEN;
-                if (this.options.onOpen) { this.options.onOpen(); }
-                this.retryStrategy.reset(); //We need to reset the counter once the connection has been re-established.
-            },
-            onEvent: (event: Event) => {
-                this.assertState(['OPEN']);
-                if (this.options.onEvent) { this.options.onEvent(event); }
-                console.assert(
-                    !this.lastEventIdReceived ||
-                    parseInt(event.eventId) > parseInt(this.lastEventIdReceived),
-                    'Expected the current event id to be larger than the previous one'
-                );
-                this.lastEventIdReceived = event.eventId;
-            },
-            onEnd: () => {
-                this.state = ResumableSubscriptionState.ENDED;
-                if (this.options.onEnd) { this.options.onEnd(); }
-            },
-            onError: (error: Error) => {
-                this.state = ResumableSubscriptionState.OPENING
-                this.retryStrategy.attemptRetry(error)
-                .then(() => {
-                  if (this.options.onRetry !== undefined) {
-                    this.options.onRetry();
-                  } else {
-                    this.tryNow();
-                  }
-                })
-                .catch(error => {
-                    this.state = ResumableSubscriptionState.ENDED;
-                    if (this.options.onError) { this.options.onError(error); }
-                })},
-            logger: this.logger
-        });
-               
-        if (this.options.tokenProvider) {
-            this.options.tokenProvider.fetchToken().then((jwt) => {
-                this.subscription.open(jwt);
-            }).catch((err) => {
-                if(this.options.onError) this.options.onError(err);
+        
+        if (this.lastEventIdReceived) {
+            newXhr.setRequestHeader("Last-Event-Id", this.lastEventIdReceived);
+        }
+        
+        this.options.tokenProvider.fetchToken()
+        .then( token => {
+            this.baseSubscription = new BaseSubscription(newXhr, {
+                path: this.options.path,
+                headers: this.options.headers,
+                jwt: token,
+                
+                onOpen: () => {
+                    this.options.onOpen();
+                    this.retryStrategy.reset(); //We need to reset the counter once the connection has been re-established.
+                },
+                onEvent: (event: SubscriptionEvent) => {
+                    this.options.onEvent(event);
+                    this.lastEventIdReceived = event.eventId;
+                },
+                onEnd: () => {
+                    this.options.onEnd();
+                },
+                onError: (error: Error) => {
+                    this.retryStrategy.attemptRetry(error)
```

Comment:
`attemptRetry` sounds like the the request is going to be attempted again here, but it looks like it then only conditionally happens if `tryNow()` is called...is that correct? If so, I think `attemptRetry` is a misleading name and instead we should use something like `shouldRetry` or `checkIfRetryable`. WDYT?